libs/wingsdk/src/target-sim/state.inflight.ts

```diff
@@ -0,0 +1,31 @@
+import { StateSchema } from "./schema-resources";
+import { IStateClient } from "./state";
+import { ISimulatorContext, ISimulatorResourceInstance } from "../simulator";
+
+export class State implements IStateClient, ISimulatorResourceInstance {
+  constructor(
+    private readonly path: string,
+    _props: StateSchema["props"],
+    private readonly context: ISimulatorContext
+  ) {}
+  public async init(): Promise<Record<string, any>> {
+    return {};
+  }
+
+  public async cleanup(): Promise<void> {}
+
+  public async set(key: string, value: any): Promise<void> {
+    // once the property is set on the object, the token will be resolved.
+    Object.defineProperty(this, key, { value });
+    this.context.setResourceAttributes(this.path, { [key]: value });
```

Comment:
What is the difference between these lines? I'm not sure I understand what the `defineProperty` line does

Reply:
The `defineProperty` was a residual from a previous implementation. Not needed whatsoever...